executor/show.go

```diff
@@ -1029,7 +1029,13 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
 			}
 		}
 		if ddl.IsAutoRandomColumnID(tableInfo, col.ID) {
-			buf.WriteString(fmt.Sprintf(" /*T![auto_rand] AUTO_RANDOM(%d) */", tableInfo.AutoRandomBits))
+			if !ctx.GetSessionVars().EnableSimplifiedShowCreateTable {
+				buf.WriteString(" /*T![auto_rand]")
+			}
+			buf.WriteString(fmt.Sprintf(" AUTO_RANDOM(%d)", tableInfo.AutoRandomBits))
+			if !ctx.GetSessionVars().EnableSimplifiedShowCreateTable {
+				buf.WriteString(" */")
+			}
```

Comment:
There are a lot of these conditional checks. Have you considered using a helper func?
```suggestion
			WriteBufIfNotSimplified(ctx, buf, " /*T![auto_rand]")
			buf.WriteString(fmt.Sprintf(" AUTO_RANDOM(%d)", tableInfo.AutoRandomBits))
			WriteBufIfNotSimplified(ctx, buf, " */")
```